### PR TITLE
refactor: client-side generated unique identifier provided as messageRequestId

### PR DESCRIPTION
**Pull request by paypal-are**

### CODE REVIEW

File: src/zoid/message/component.js [hunk 1]
```diff
@@ -177,7 +177,13 @@ export default createGlobalVariableGetter('__paypal_credit_message__', () =>
 
                     return ({ meta, activeTags, deviceID }) => {
                         const { account, merchantId, index, modal, getContainer } = props;
-                        const { messageRequestId, trackingDetails, offerType, ppDebugId } = meta;
+                        // Message request ID generated server-side
+                        const {
+                            messageRequestId: serverMessageRequestId,
```
Comment (gillycheesesteak) [inline src/zoid/message/component.js]: We should consider moving the `serverMessageRequestId` into the encrypted tracking payload so we do not have to access it explicitly on the client

**Review by danzhaas — APPROVED** [review]

**Review by dj110919871 — APPROVED** [review]

File: src/zoid/message/component.js [hunk 2]
```diff
@@ -176,9 +176,11 @@ export default createGlobalVariableGetter('__paypal_credit_message__', () =>
                 value: ({ props }) => {
                     const { onReady } = props;
 
-                    return ({ meta, activeTags, deviceID }) => {
+                    return ({ meta, activeTags, deviceID, messageRequestId }) => {
```
Comment (gillycheesesteak) [inline src/zoid/message/component.js]: If this value doesn't change on re-renders, can we pull it from `props` below?

File: src/components/message/Message.js [hunk 3]
```diff
@@ -147,17 +147,13 @@ const Message = function({ markup, meta, parentStyles, warnings }) {
...
-                                    meta: data.meta ?? meta,
-                                    activeTags: getActiveTags(button),
-                                    // Utility will create iframe deviceID if it doesn't exist.
-                                    deviceID: getOrCreateStorageID()
-                                });
-                            }
+                            onReady({
+                                meta: data.meta ?? meta,
+                                activeTags: getActiveTags(button),
+                                messageRequestId,
```
Comment (gillycheesesteak) [inline src/components/message/Message.js]: Should we generate a new `messageRequestId` here when the message updates?

**Review by gillycheesesteak — APPROVED** [review]

**Pull request merged**